Update sGho Docs (#30)
aave-dao/gho-origin: docs/sgho.md

Reviewer: i d kep this section simple in this way, and add information about the compounding effect in the precision md file.
(link is fixed now)
```suggestion
sGHO uses high-precision arithmetic to ensure accurate yield calculations and prevent precision loss during share/asset conversions.

The following are key considerations for arithmetic precision in math operations:
- **Yield Index**: Stored with RAY precision (1e27) to maintain accuracy over long periods
- **Rate Calculations**: Annual rates converted to per-second rates with sufficient precision
- **Share Conversions**: Asset-to-share and share-to-asset conversions use high-precision math
- **Accumulated Interest**: Linear interest accumulation calculated with RAY precision

For a comprehensive analysis of precision handling, edge cases, and mathematical considerations, see the [detailed precision analysis document](./sgho-precision-analysis/precision.md)**

```

### Key Parameters

- **Target Rate**: Annual percentage rate in basis points (max 50% = 5000). Maximum rate can be higher with frequent updates.
- **Rate Per Second**: Rate at which the index will increase for each second passed (calculated from the set Target Rate)
- **Yield Index**: Index used for share/asset conversions

## Role Management

> - `PAUSE_GUARDIAN_ROLE` : This role has permissions to pause/unpause any action related to sGho shares:, including deposits, withdrawals and transfers.

Reviewer: ```suggestion
- `PAUSE_GUARDIAN_ROLE` : This role has permissions to pause/unpause any action related to sGho shares including deposits, withdrawals and transfers.
```

> - `PAUSE_GUARDIAN_ROLE` : This role has permissions to pause/unpause sGho deposits and withdrawals. Transfers are also frozen when the contract is paused.

Reviewer: ```suggestion
- `PAUSE_GUARDIAN_ROLE` : This role has permissions to pause/unpause any action related to sGho shares:, including deposits, withdrawals and transfers.
```

- `TOKEN_RESCUER_ROLE` : This role has permissions to rescue tokens held on the contract
- `YIELD_MANAGER_ROLE` : This role has permissions to update the yield target rate and the supply cap.

## Security Considerations

### Built-in Protections

- **Supply Cap**: Limits maximum vault capacity

> - **Rate Limits**: Maximum 50% annual rate to prevent excessive yield. Maximum rate can be higher with frequent updates.

Reviewer: do not think this is needed. Better to expand in case we want to talk about the compounding effect

## Precision

> 

Reviewer: ```suggestion

```

> ### Overview

Reviewer: ```suggestion
## Math
```
